src/services/redisSessionStorage.ts

```diff
@@ -107,8 +109,7 @@ export default class RedisSessionStorage extends RedisStorageUtils
     const errorOrSession = await this.loadSessionBySessionToken(token);
 
     if (isLeft(errorOrSession)) {
-      const error = errorOrSession.value;
-      return left(error);
+      return left(errorOrSession.value); // TODO: Error not propagated
```

Comment:
@cloudify on this scenario the error is not propagated to the client and i think is not logged completely server side.
When this method is called the `fulfill` function ignore the error and set user to `false`:
```
sessionStorage.getBySessionToken(token as SessionToken).then(
  (errorOrUser: Either<Error, User>) => {
    fulfill(errorOrUser, done);
  },
  () => {
    done(undefined, false);
  }
);
```
Reference to the file [here](https://github.com/teamdigitale/io-backend/blob/166785497-better-redis-logs/src/strategies/bearerTokenStrategy.ts#L60)

Reply:
yeah I think in that case the user gets a 401 - `getBySessionToken ` should return a`Promise<Either<Error, Option<User>>>` to be able to distinguish errors from missing sessions

Reply:
once you have that, you can provide an error reason in the authentication callback in the bearer token strategy, see [here](http://www.passportjs.org/docs/configure/)

Reply:
Fixed in 5d94c37 and bda2514